broadinstitute/scp-ingest-pipeline
# Enable local development via Docker compose (SCP-4883)

This improves our ability to develop Ingest Pipeline in Apple silicon Macs (e.g. M1), Broad's standard CPU for some time.  In particular, it aims to onboard Michelle to fully contribute to Single Cell Portal.  

The new Docker compose setup reuses patterns from `single_cell_portal_core`.  The updated README here has instructions.  There is known related work (per TODO comments), but the current state seems like a decent working start.

We know that we can _test_ on M1 via local Docker, but my understanding is that no one has been able to _develop_ Ingest Pipeline on an M1.  More background discussion on this is [in #scp-implementation in Slack](https://broadinstitute.slack.com/archives/CBEHTH601/p1673967512735419).  Jon, Michelle and I mobbed on this Thursday; Jon's navigation was essential.

To test:
* Run `./scripts/docker-compose-setup.sh -t ~/.github-token`; where `-t` takes as a value your GitHub token for Vault
* Ensure you see output like the following, and that you enter a Bash shell:
```
### SETTING VAULT TOKEN ###
### SETTING UP ENVIRONMENT ###
Success! You are now authenticated. The token information displayed below
is already stored in the token helper. You do NOT need to run "vault login"
again. Future Vault requests will automatically use this token.

<Some lines excerpted>

Status: Image is up to date for gcr.io/broad-singlecellportal-staging/scp-ingest-pipeline-development:661144f
gcr.io/broad-singlecellportal-staging/scp-ingest-pipeline-development:661144f
root@ac32579d655e:/scp-ingest-pipeline/ingest# 
```

This relates to SCP-4883.
